src/IR/exprs.ts

```diff
@@ -173,7 +173,9 @@ export function polygolfOp(op: OpCode, ...args: Expr[]): Expr {
         } else newArgs.push(arg);
       }
       args = newArgs;
-      if (
+      if (op === "mul" && args.length === 2 && isIntLiteral(args[0], 1n)) {
```

Comment:
```suggestion
      if (op === "mul" && args.length > 1 && isIntLiteral(args[0], 1n)) {
```
Removes redundant multiplication by 1 for arbitrarily sized multiplication chains, as currently
```
$a:-5..5 <- 0;
$b:-5..5 <- 0;
$c <- (* (- $a) (- $b));
$d <- (* 1 $a $b);
```
compiles to 
```
$a:-5..5 <- 0;
$b:-5..5 <- 0;
$c <- (* 1 $a $b);
$d <- (* 1 $a $b);
```